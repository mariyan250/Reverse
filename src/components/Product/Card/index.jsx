// React and Style
import React from 'react';
import styles from './index.module.scss';

// Router
import { Link } from 'react-router-dom';

// Translation
import { useTranslation } from 'react-i18next';

function Item({ id, images, name, price, stars }) {
  const { t } = useTranslation();

  return (
    <li className={`${styles.item} app-bg-white overflow-hidden rounded-lg`}>
      <Link to={`/item/${id}`} className="app-text-secondary">
        <div className={`${styles.container} d-flex position-relative justify-content-center`}>
          <img
            src={images[0]}
            alt={t('search_screen.item.img.alt')}
            className={`${styles.img} img-fluid position-absolute p-5`}
          />
        </div>

        <div className="p-2 h-100">
          <div className="p-3 app-bg-gray-darkest rounded-lg">
            <p className={`${styles['item-name']} mb-2`}>{name}</p>

            <div className={`${styles['item-stars']} mb-3`}>
              {Array(stars)
                .fill()
                .map(() => (
                  <i className="fas fa-star"></i>
                ))}
            </div>

            <div className="d-flex justify-content-between align-items-center">
              <p class={`${styles['item-price']} app-text-primary`}>
                {price} {t('general.currency')}
              </p>

              <button className="border-0 py-3 px-4 py-md-2 px-md-3 app-text-white rounded-lg app-bg-primary">
                Add to cart
              </button>
            </div>
          </div>
        </div>
      </Link>
    </li>
  );
}

export default Item;
