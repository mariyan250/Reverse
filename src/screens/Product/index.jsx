// React and style
import React, { useEffect } from 'react';
import './index.scss';

// Redux
import { connect } from 'react-redux';
import { getPost } from 'store/actions/post';

// Translations
import { useTranslation } from 'react-i18next';

// Image Gallery
import ImageGallery from 'react-image-gallery';

// Components
import NotFound from 'screens/NotFound';

function Product({ getPost, post, match }) {
  const { t } = useTranslation();

  useEffect(() => {
    if (match.params) {
      getPost(match.params.id);
    }
  }, [getPost, post, match]);

  const getImages = (data) => {
    return data.images.map((val) => {
      return {
        original: val,
        thumbnail: val,
      };
    });
  };

  if (!post) {
    return <NotFound />;
  }

  return (
    <section className="product p-lg-4" data-testid="product">
      <article className="gallery px-3">
        {post.images && (
          <ImageGallery
            items={getImages(post)}
            showPlayButton={false}
            showBullets={true}
            showThumbnails={false}
            showFullscreenButton={false}
          />
        )}
      </article>

      <article className="item-description p-4 pb-5 p-lg-5">
        <h1 className="mb-3 mt-4 mt-md-0">{post.name}</h1>
        <p className="mb-3 mt-4">
          Lorem ipsum dolor, sit amet consectetur adipisicing elit. Dolorem voluptatibus, laborum ut
          aliquid optio architecto quos libero sint odit earum fugiat alias. Consequuntur quam
          eaque, dolorum necessitatibus quisquam iste, harum pariatur quos qui placeat nam maiores
          debitis similique fugiat itaque, dolore doloribus magnam tempore natus ipsam aspernatur
          maxime molestias sequi?
        </p>

        <section className="item-description-second d-md-flex justify-content-between my-4 my-md-0 mt-md-5">
          <p>
            <i className="fas fa-map-marker-alt mr-3" />
            {post.location}
          </p>

          <p>
            <i className="fas fa-clock mr-3" />
            {post.date}
          </p>

          <p className="mb-5">
            <i className="fas fa-user mr-3" />
            {post.seller}
          </p>
        </section>

        <section className="item-description-nav app-bg-white d-flex justify-content-around justify-content-sm-between align-items-center py-1 m-sm-0 mt-4 mt-md-0 w-100">
          <h5 className="text-center my-2 my-md-0">
            {post.price} {t('general.currency')}
          </h5>
          <span className="app-bg-primary app-text-white rounded py-3 px-5 px-md-4 py-md-2 mr-2">
            {t('product.call_button')}
            <i className="fas fa-phone ml-3" />
          </span>
        </section>
      </article>
    </section>
  );
}

const mapStateToProps = (state) => ({ post: state.posts.item });

export default connect(mapStateToProps, { getPost })(Product);
