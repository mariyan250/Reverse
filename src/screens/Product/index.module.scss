@import '../../index.scss';

.product {
  display: grid;

  .image-gallery-thumbnails {
    display: none;

    @include media-breakpoint-md-up {
      display: block;
    }
  }

  @include media-breakpoint-lg-up {
    grid-template-columns: to-rem(600px) 1fr;
  }

  @include media-breakpoint-xl-up {
    grid-template-columns: to-rem(650px) 1fr;
  }

  .description {
    span {
      font-size: to-rem(16px);
    }

    .down {
      p {
        font-size: to-rem(18px);
        margin: to-rem(24px) 0;

        @include media-breakpoint-md-up {
          margin: 0;
        }
      }
    }

    .nav {
      position: fixed;
      bottom: 0;
      left: 0;
      box-shadow: 0px 0px 3px gray;

      h5 {
        flex: 1;

        @include media-breakpoint-sm-up {
          flex: none;
        }
      }

      @include media-breakpoint-sm-up {
        position: relative;
        box-shadow: none;
      }
    }
  }
}
