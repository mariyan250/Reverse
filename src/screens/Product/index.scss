@import '~react-image-gallery/styles/scss/image-gallery.scss';
@import '../../index.scss';

.item-screen {
  display: grid;

  .image-gallery-thumbnails {
    display: none;

    @include media-breakpoint-md-up {
      display: block;
    }
  }

  @include media-breakpoint-lg-up {
    grid-template-columns: to-rem(600px) 1fr;
  }

  @include media-breakpoint-xl-up {
    grid-template-columns: to-rem(700px) 1fr;
  }
}
