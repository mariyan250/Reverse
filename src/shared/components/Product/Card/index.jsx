// React and Style
import React from 'react';
import './index.scss';

// Router
import { Link } from 'react-router-dom';

// Translation
import { useTranslation } from 'react-i18next';

// TODO use translation for currency
function Item({ id, images, name, location, date, price }) {
  const { t } = useTranslation();

  return (
    <li className="item app-bg-white box-shadow overflow-hidden">
      <Link to={`/item/${id}`} className="app-text-secondary">
        <section className="img-container d-flex overflow-hidden position-relative">
          <img
            src={images[0]}
            alt={t('search_screen.item.img.alt')}
            className="img-fluid position-absolute p-3"
          />
        </section>

        <section className="py-3 px-4 h-100 app-bg-gray">
          <p className="product-name text-center mb-4">{name}</p>
          <p className="product-location mb-3">
            <i className="fas fa-map-marker-alt mr-2" />
            {location}
          </p>
          <p className="product-date mb-4">
            <i className="fas fa-clock mr-2" />
            {date}
          </p>
          <p className="product-price">
            {price} {t('general.currency')}
          </p>
        </section>
      </Link>
    </li>
  );
}

export default Item;
