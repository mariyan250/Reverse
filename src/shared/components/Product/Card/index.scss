@import '../../../../index.scss';

.item {
  .img-container {
    img {
      object-fit: cover;
      top: 50%;
      transform: translateY(-50%);
    }

    &::before {
      display: 'block';
      content: '';
      padding-top: 100%;
    }
  }

  section {
    .product-name {
      font-size: to-rem(16px);
      font-weight: 600;
    }

    .product-price {
      font-size: to-rem(18px);
    }
  }
}
