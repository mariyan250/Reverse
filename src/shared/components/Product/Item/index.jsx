// React and Style
import React from 'react';
import './index.scss';

// Router
import { Link } from 'react-router-dom';

// Translation
import { useTranslation } from 'react-i18next';

// TODO use translation for currency
function Item({ id, images, name, location, date, price }) {
  const { t } = useTranslation();

  return (
    <li className="item app-bg-white box-shadow">
      <Link to={`/item/${id}`} className="app-text-secondary">
        <section className="img-container d-flex overflow-hidden position-relative">
          <img
            src={images[0]}
            alt={t('search_screen.item.img.alt')}
            className="img-fluid position-absolute"
          />
        </section>

        <section className="p-3">
          <h6 className="my-3 text-center">{name}</h6>
          <p className="mt-4 mb-2 font-weight-light">
            <i className="fas fa-map-marker-alt mr-2" />
            {location}
          </p>
          <article className="d-flex justify-content-between align-items-end">
            <p>
              <i className="fas fa-clock mr-2" />
              {date}
            </p>
            <h4>{price} $</h4>
          </article>
        </section>
      </Link>
    </li>
  );
}

export default Item;
