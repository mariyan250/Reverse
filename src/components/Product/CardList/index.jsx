// React and style
import React from 'react';
import styles from './index.module.scss';

// Components
import Item from '../Card';

function ItemList({ items }) {
  return (
    <ul className={styles.list}>
      {items && items.map((item) => <Item {...item} key={item.id} />)}
    </ul>
  );
}

export default ItemList;
