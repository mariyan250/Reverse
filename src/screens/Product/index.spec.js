import React from 'react';
import { render, waitForElement, cleanup } from 'tests/test-utils';
import Product from './index';

describe('ItemScreen', () => {
  let initialState;

  beforeAll(() => {
    initialState = {};
  });

  afterEach(cleanup);

  describe('Rendering', () => {
    it('should render the component with initial state', async () => {
      const { container, getByTestId } = render(
        <Product match={{ params: { id: '3' } }} />,
        initialState,
      );
      await waitForElement(() => getByTestId('product'));
      expect(container).toMatchSnapshot();
    });
  });
});
