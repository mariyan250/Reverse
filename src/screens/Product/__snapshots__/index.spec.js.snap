// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemScreen Rendering should render the component with initial state 1`] = `
<div>
  <section
    class="product p-lg-4"
    data-testid="product"
  >
    <article
      class="gallery px-3"
    >
      <div
        aria-live="polite"
        class="image-gallery"
      >
        <div
          class="image-gallery-content bottom"
        >
          <div
            class="image-gallery-slide-wrapper bottom"
          >
            <button
              aria-label="Previous Slide"
              class="image-gallery-icon image-gallery-left-nav"
              type="button"
            >
              <svg
                class="image-gallery-svg"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1"
                viewBox="6 0 12 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="15 18 9 12 15 6"
                />
              </svg>
            </button>
            <button
              aria-label="Next Slide"
              class="image-gallery-icon image-gallery-right-nav"
              type="button"
            >
              <svg
                class="image-gallery-svg"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1"
                viewBox="6 0 12 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="9 18 15 12 9 6"
                />
              </svg>
            </button>
            <div
              class="image-gallery-swipe"
            >
              <div
                class="image-gallery-slides"
              >
                <div
                  aria-label="Go to Slide 1"
                  class="image-gallery-slide  center "
                  role="button"
                  style="transform: translate3d(0%, 0, 0);"
                  tabindex="-1"
                >
                  <div>
                    <img
                      class="image-gallery-image"
                      src="https://s13emagst.akamaized.net/products/17043/17042927/images/res_b0edd1ced6577580eaec29aad246405d.jpg"
                    />
                  </div>
                </div>
                <div
                  aria-label="Go to Slide 2"
                  class="image-gallery-slide  right "
                  role="button"
                  style="transform: translate3d(100%, 0, 0);"
                  tabindex="-1"
                >
                  <div>
                    <img
                      class="image-gallery-image"
                      src="https://s13emagst.akamaized.net/products/17043/17042927/images/res_a395977c320f9b8440947b9cb2ef40af.jpg"
                    />
                  </div>
                </div>
                <div
                  aria-label="Go to Slide 3"
                  class="image-gallery-slide  left "
                  role="button"
                  style="transform: translate3d(-100%, 0, 0);"
                  tabindex="-1"
                >
                  <div>
                    <img
                      class="image-gallery-image"
                      src="https://s13emagst.akamaized.net/products/17043/17042927/images/res_15bda9b038fc372c49488bb599810eda.jpg"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="image-gallery-bullets"
            >
              <div
                aria-label="Bullet Navigation"
                class="image-gallery-bullets-container"
                role="navigation"
              >
                <button
                  aria-label="Go to Slide 1"
                  aria-pressed="true"
                  class="image-gallery-bullet active"
                  type="button"
                />
                <button
                  aria-label="Go to Slide 2"
                  aria-pressed="false"
                  class="image-gallery-bullet"
                  type="button"
                />
                <button
                  aria-label="Go to Slide 3"
                  aria-pressed="false"
                  class="image-gallery-bullet"
                  type="button"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </article>
    <article
      class="item-description p-4 pb-5 p-lg-5"
    >
      <h1
        class="mb-3 mt-4 mt-md-0"
      >
        Iphone X, 128SD / 256SD, Black and White
      </h1>
      <p
        class="mb-3 mt-4"
      >
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Dolorem voluptatibus, laborum ut aliquid optio architecto quos libero sint odit earum fugiat alias. Consequuntur quam eaque, dolorum necessitatibus quisquam iste, harum pariatur quos qui placeat nam maiores debitis similique fugiat itaque, dolore doloribus magnam tempore natus ipsam aspernatur maxime molestias sequi?
      </p>
      <section
        class="item-description-second d-md-flex justify-content-between my-4 my-md-0 mt-md-5"
      >
        <p>
          <i
            class="fas fa-map-marker-alt mr-3"
          />
          Blagoevgrad
        </p>
        <p>
          <i
            class="fas fa-clock mr-3"
          />
          8 July - 6:00
        </p>
        <p
          class="mb-5"
        >
          <i
            class="fas fa-user mr-3"
          />
          Mariyan V.
        </p>
      </section>
      <section
        class="item-description-nav app-bg-white d-flex justify-content-around justify-content-sm-between align-items-center py-1 m-sm-0 mt-4 mt-md-0 w-100"
      >
        <h5
          class="text-center my-2 my-md-0"
        >
          1200
           
          general.currency
        </h5>
        <span
          class="app-bg-primary app-text-white rounded py-3 px-5 px-md-4 py-md-2 mr-2"
        >
          product.call_button
          <i
            class="fas fa-phone ml-3"
          />
        </span>
      </section>
    </article>
  </section>
</div>
`;
